engine/gamesys/src/gamesys/components/comp_sprite.cpp

```diff
@@ -1002,14 +1053,15 @@ namespace dmGameSystem
         return !data->m_UsesGeometries || data->m_Geometries[0]->m_TrimMode == dmGameSystemDDF::SPRITE_TRIM_MODE_OFF;
     }
 
-    static void ResolvePositionAndUVDataFromGeometry(TexturesData* data, dmArray<float>* scratch_pos, dmArray<float>* scratch_uvs, float* scratch_uv_ptrs[MAX_TEXTURE_COUNT],
-                                                            float scale_x, float scale_y, int reverse)
+    static void ResolvePositionAndUVDataFromGeometry(TexturesData* data,
+        dmArray<Vector4>& scratch_pos, dmArray<float>* scratch_uvs,
+        float* scratch_uv_ptrs[MAX_TEXTURE_COUNT], float scale_x, float scale_y, int reverse)
```

Comment:
The scratch pos is a vec4 now since we require that all transforms are done before passing in to the vertex write function. Our Vector3 format is actually a Vector4 so it's just simpler to use a Vector4 through and through.